src/confirm_subscription.rs

```diff
@@ -0,0 +1,74 @@
+#[ft_sdk::processor]
+pub fn confirm_subscription(
+    mut conn: ft_sdk::Connection,
+    ft_sdk::Query(code): ft_sdk::Query<"code">,
+    ft_sdk::Query(email): ft_sdk::Query<"email">,
+    ft_sdk::Query(next): ft_sdk::Query<"next", Option<String>>,
+) -> ft_sdk::processor::Result {
+    use diesel::prelude::*;
+
+    if !validator::ValidateEmail::validate_email(&email) {
+        return Err(ft_sdk::single_error("email", "Invalid email format.").into());
+    }
+
+    #[derive(diesel::QueryableByName)]
+    #[diesel(table_name = ft_sdk::auth::fastn_user)]
+    struct UserData {
+        id: i64,
+        data: String,
+    }
+
+    let (user_id, data): (i64, serde_json::Value)  = match diesel::sql_query(
+        r#"
+        SELECT id, data from fastn_user where json_extract(data, '$.subscription.confirmation_key') = $1 LIMIT 1;
+        "#,
+    )
+    .bind::<diesel::sql_types::Text, _>(&code)
+    .get_result::<UserData>(&mut conn)
+    {
+        Ok(d) => (d.id, serde_json::from_str(&d.data)?),
+        Err(e) => return Err(e.into()),
+    };
+
+    let data = {
+        let mut data = data;
+
+        data = mark_user_verified(data);
+
+        data
+    };
+    let data = serde_json::to_string(&data)?;
+
+    diesel::update(
+        ft_sdk::auth::fastn_user::table.filter(ft_sdk::auth::fastn_user::id.eq(user_id)),
+    )
+    .set((
+        ft_sdk::auth::fastn_user::data.eq(&data),
+        ft_sdk::auth::fastn_user::updated_at.eq(ft_sdk::env::now()),
+    ))
+    .execute(&mut conn)?;
```

Comment:
Use `ft_sdk::auth::provider::update_user`

Reply:
that can't be used here. Tha uses `HashMap::insert` on `data` which will not do deep merge. This is correct, for now.

Reply:
@siddhantk232 in that case we should use another method to update user from ft-sdk. We have to create an appropriate one, as others will also use it, and we can try to make sure ft-sdk upholds our assumptions.